src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/dbscripts/query_references.sql

```diff
@@ -27,5 +27,5 @@ SELECT
 FROM "{{baseDatabase}}"."{{#if vars.tableName}}{{vars.tableName}}{{else}}DBQLObjTbl{{/if}}"
 {{#if queryLogsVariables.timeRange}}
 WHERE "{{baseDatabase}}"."{{#if vars.tableName}}{{vars.tableName}}{{else}}DBQLObjTbl{{/if}}"."CollectTimeStamp"
-  BETWEEN TIMESTAMP '{{queryLogsVariables.timeRange.startTimestamp}}' AND TIMESTAMP '{{queryLogsVariables.timeRange.endTimestamp}}'
+  BETWEEN TIMESTAMP '{{queryLogsVariables.timeRange.startTimestamp}}+00:00' AND TIMESTAMP '{{queryLogsVariables.timeRange.endTimestamp}}+00:00'
```

Comment:
It turns out that although "Z" doesn't guarantee it consistently working for Teradata, "xxx" will do the magic.

Good idea, it gives us more freedom if we happen to want to use non-UTC inputs in the future.

The reason it is required is that the TIMESTAMP literal interprets the input with adjustments based on the time zone of the database. (I didn't notice it until I found that "9999-12-31 23:59:59.99" incurs a "Invalid TimeStamp Literal" error on the Teradata side while "9999-12-31 23:59:59.99+00:00" does not - that is because Teradata tacitly tries to add displacement on top of the already upper-bounded value.